algo/src/main/java/com/hartwig/serve/sources/ckb/CkbExtractor.java

```diff
@@ -128,7 +128,8 @@ private ExtractionResult getExtractionResult(@NotNull CkbEntry entry) {
 
             if (generateKnownEvents) {
                 extractionResultBuilder.knownHotspots(convertToKnownHotspots(extractionOutput.hotspots(), event, variant))
-                        .knownCodons(convertToKnownCodons(extractionOutput.codons(), variant))
+                        .knownCodons(convertToKnownCodons(actionableEntries.isEmpty() ? Collections.emptyList() : extractionOutput.codons(),
+                                variant))
```

Comment:
This change makes the output match the current research output, but is it desirable? In master, the input list of codons is initialized for each entry of `actionableEntries` or empty if there are no entries. Omitting input codons when there are no actionable entries reduces the known codon list from 124 to 30 (e.g. 6 codons for gene ERBB2, which is omitted entirely with this change).

Reply:
This means that we only consider a codon to be a known codon if the codon has associated clinical evidence? 

I think we've made that choice since CKB is actionability-focused, and we only want to consider a codon to be known (=pathogenic) if its evidence passes our filtering.

Reply:
Thanks for confirming!